From 0c295ff25ec9af8ecca813bc4809d83c8d93ef19 Mon Sep 17 00:00:00 2001
From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Fri, 27 Mar 2009 14:59:14 +0300
Subject: [PATCH 76/82] cfq link cfq_bc_data without bc io sched

Fixes oops at first IO with CONFIG_BC_IO_SCHED=n.

The cfq_set_request wants to get ub by cfqq->cfq_bc->ub_iopriv,
so save ref to ub0 there.

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 block/cfq-iosched.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/block/cfq-iosched.c b/block/cfq-iosched.c
index 497b791..264a086 100644
--- a/block/cfq-iosched.c
+++ b/block/cfq-iosched.c
@@ -2163,6 +2163,8 @@ static void *cfq_init_queue(struct request_queue *q)
 	INIT_LIST_HEAD(&cfqd->act_cfq_bc_head);
 #ifndef CONFIG_BC_IO_SCHED
 	cfq_init_cfq_bc(&cfqd->cfq_bc);
+	cfqd->cfq_bc.cfqd = cfqd;
+	cfqd->cfq_bc.ub_iopriv = &ub0.iopriv;
 	/*
 	 *  Adding ub0 to active list in order to serve force dispatching
 	 *  case uniformally. Note, that nobody removes ub0 from this list.
-- 
1.6.2.2

